pkg/config/viperconfig/viper.go

```diff
@@ -82,7 +83,6 @@ func (c *safeConfig) Set(key string, newValue interface{}, source model.Source)
 	// modify the config then release the lock to avoid deadlocks while notifying
 	var receivers []model.NotificationReceiver
 	c.Lock()
```

Comment:
I think we would want to do a deferred unlock here to match the behavior we have in NTM now?

Reply:
And then also we'd want to remove the declaration of `receivers`, and just `return` if the value changed... basically just make this look more like what we're doing in NTM.